accountStorageExporter/main.go

```diff
@@ -0,0 +1,300 @@
+		if err != nil {
+			return nil, fmt.Errorf("%w creating a log file", err)
+		}
+	}
+
+	err = logger.SetDisplayByteSlice(logger.ToHex)
+	log.LogIfError(err)
+	logger.ToggleLoggerName(flagsConfig.EnableLogName)
+	logLevelFlagValue := flagsConfig.LogLevel
+	err = logger.SetLogLevel(logLevelFlagValue)
+	if err != nil {
+		return nil, err
+	}
+
+	if flagsConfig.DisableAnsiColor {
+		err = logger.RemoveLogObserver(os.Stdout)
+		if err != nil {
+			return nil, err
+		}
+
+		err = logger.AddLogObserver(os.Stdout, &logger.PlainFormatter{})
+		if err != nil {
+			return nil, err
+		}
+	}
+	log.Trace("logger updated", "level", logLevelFlagValue, "disable ANSI color", flagsConfig.DisableAnsiColor)
+
+	return fileLogging, nil
+}
+
+func getMaxDBValue(parentDir string) (int, error) {
+	contents, err := ioutil.ReadDir(parentDir)
+	if err != nil {
+		return 0, err
+	}
+
+	directories := make([]string, 0)
+	for _, c := range contents {
+		if !c.IsDir() {
+			continue
+		}
+
+		_, ok := big.NewInt(0).SetString(c.Name(), 10)
+		if !ok {
+			log.Debug("DB directory found that will not be taken into account", "name", c.Name())
+			continue
+		}
+
+		directories = append(directories, c.Name())
+	}
+
+	numDirs := 0
+	for i := 0; i < maxDirs; i++ {
+		expectedDir := fmt.Sprintf("%d", i)
+		if !contains(directories, expectedDir) {
+			break
+		}
+
+		numDirs++
+	}
+
+	if numDirs == 0 {
+		return 0, fmt.Errorf("missing ordered directories in %s, like 0, 1 and so on", parentDir)
+	}
+	if numDirs != len(directories) {
+		return 0, fmt.Errorf("unordered directories in %s, like 0, 1 and so on", parentDir)
+	}
+
+	return numDirs - 1, nil
+}
+
+func contains(haystack []string, needle string) bool {
+	for _, h := range haystack {
+		if h == needle {
+			return true
+		}
+	}
+
+	return false
+}
+
+func exportStorage(address string, flags config.ContextFlagsConfig, mainRootHash []byte, maxDBValue int) error {
+	addressConverter, err := pubkeyConverter.NewBech32PubkeyConverter(addressLength, log)
+	if err != nil {
+		return err
+	}
+
+	tr, err := getTrie(flags, maxDBValue)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		errNotCritical := tr.Close()
+		log.LogIfError(errNotCritical)
+	}()
+
+	accDb, err := newAccountsAdapter(tr)
+	if err != nil {
+		return err
+	}
+
+	err = accDb.RecreateTrie(mainRootHash)
+	if err != nil {
+		return err
+	}
+
+	addressBytes, err := addressConverter.Decode(address)
+	if err != nil {
+		return err
+	}
+
+	account, err := accDb.GetExistingAccount(addressBytes)
+	if err != nil {
+		return err
+	}
+
+	userAccount, ok := account.(state.UserAccountHandler)
+	if !ok {
+		return fmt.Errorf("cannot cast AccountHandler to UserAccountHandler")
+	}
+
+	leavesCh, err := userAccount.DataTrie().GetAllLeavesOnChannel(mainRootHash)
+	if err != nil {
+		return err
+	}
+
+	keyValueMap := make(map[string]string)
+	for leaf := range leavesCh {
+		keyValueMap[hex.EncodeToString(leaf.Key())] = hex.EncodeToString(leaf.Value())
+	}
+
```

Comment:
For consistency, I think we should sort the map. Subsequent calls can be easily diff checked.

Reply:
Not a bad idea, but I cannot sort the map. I can only print it in a given order (by using an additional sorted slice for the keys), but I cannot guarantee the file's output. Also, I wouldn't change the output to array or something else